tc_driver/Cases.jl

```diff
@@ -226,21 +266,26 @@ end
 function initialize_profiles(::Soares, grid::Grid, param_set, state; kwargs...)
     aux_gm = TC.center_aux_grid_mean(state)
     prog_gm = TC.center_prog_grid_mean(state)
-    ρ_c = prog_gm.ρ
     FT = TC.float_type(state)
     prof_q_tot = APL.Soares_q_tot(FT)
     prof_θ_liq_ice = APL.Soares_θ_liq_ice(FT)
     prof_u = APL.Soares_u(FT)
     prof_tke = APL.Soares_tke(FT)
+    # Solving the initial value problem
+    p_0::FT = FT(1000 * 100) # TODO - duplicated from surface_ref_state
+    z_0::FT = FT(0)          # TODO - where to get it from? Those are also our CLI and namelist args...
```

Comment:
maybe 
```
kf_surf = TC.kf_surface(grid)
kf_toa = TC.kf_top_of_atmos(grid)
z_0::FT = grid.z[kf_surf]
z_max::FT = grid.z[kf_toa]
```
would work here

Reply:
Thanks @yairchn . This works, though it feels like running in circles ;)
```
    z_0::FT = grid.zf[TC.kf_surface(grid)].z                                    
    z_max::FT = grid.zf[TC.kf_top_of_atmos(grid)].z 
```